File: leetcode/rust-leetcode/src/7.reverse-integer.rs
impl Solution {
    pub fn reverse(x: i32) -> i32 {
        let mut x = x;
        let mut ans = 0;
        let mut sign = 1;
        if x < 0 {
            x = -x;
            sign = -1;
        }
        while x > 0 {
            if ans > i32::MAX/10 {
                return 0;
            }
            ans = (ans * 10) + (x % 10);
            x = x/10;
        }

        ans * sign
    }
}

File: leetcode/python/7.reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        ans = 0
        sign = 1
        if x < 0:
            sign = -1
            x = -x
        while x > 0:
            ans = ans * 10 + (x % 10)
            x //= 10
        ans = ans * sign
        if ans >= pow(2, 31) or ans < -pow(2, 31):
            return 0
        return ans
